### ISSUE TITLE
Add support for org-mode files

### ISSUE THREAD

**Issue by sirno** [valentjn/vscode-ltex#277]
**Describe the bug**
Ltex will not start with org files although enabled.

**Steps to reproduce**
1. install `vscode-org-mode` extension 
2. Add setting:
```
{
"ltex.enabled": [
    "bibtex",
    "latex",
    "markdown",
    "rsweave",
    "org"
  ]
}
```
3. Open a file with `*.org`

**Expected behavior**
Ltex starts and checks document

**Sample document**
Any org document.

**LTeX configuration**

<details>

```
{
"ltex.enabled": [
    "bibtex",
    "latex",
    "markdown",
    "rsweave",
    "org"
  ]
}
```

</details>

**Comment by neoscaler** [comment]
Can confirm this, seems like a general problem. I am trying to check Asciidoc (identifier 'asciidoc') files with Ltex. Same as described above. Checking MD works.

Settings:

```
    "ltex.enabled": [
        "bibtex",
        "latex",
        "markdown",
        "rsweave",
        "asciidoc"
    ],
```

**Comment by valentjn** [comment]
Maybe the docs of `ltex.enabled` are a bit misleading/incomplete here. Of course that setting only works for languages that are actually supported by LT<sub>E</sub>X. You can't just supply an arbitrary language and expect LT<sub>E</sub>X to be able to parse it out-of-the-box. All currently supported languages are already given in the default of `ltex.enabled`: `bibtex`, `latex`, `markdown`, and `rsweave`. See also the [feature list in the README](https://valentjn.github.io/vscode-ltex/index.html#features): &ldquo;Checks L<sup>A</sup>T<sub>E</sub>X and Markdown documents.&rdquo; So, this issue is mostly a feature request, and partly a documentation bug.

To be able to work with org-mode files, we need an org-mode parser written in Java; the parser needs to preserve source code positions. I'll look into this. @NeoScaler Please open a separate feature request for Asciidoc.

**Comment by valentjn** [comment]
The available options of org-mode parsers written in Java and in active development seem to be [orgzly/org-java](https://github.com/orgzly/org-java) and [200ok-ch/org-parser](https://github.com/200ok-ch/org-parser). The former is licensed under GPLv3 and the latter under AGPLv3, both without linking exceptions (such as the [Classpath exception](https://en.wikipedia.org/w/index.php?title=GPL_linking_exception&oldid=1016146679#The_Classpath_exception)). Unfortunately, these licenses are incompatible with the MPL-2.0 LT<sub>E</sub>X is using. I'm not sure how you can possibly license a library (it's a whole different story for standalone programs) under the GPL and not under the LGPL, as that essentially means that only GPL libraries/programs can use that library.

So, until somebody comes up with a different parser we can use (and which preserves source code positions), this feature is going to be out-of-scope.

Mentioning #91 as fixing that would help here as well.

**Comment by valentjn** [comment]
FWIW, it's possible to check arbitrary files as plain text by adding the code language ID such as `"org"` to [`ltex.enabled`](https://valentjn.github.io/vscode-ltex/docs/settings.html#ltexenabled) and activating the extension by running [`LTeX: Activate Extension`](https://valentjn.github.io/vscode-ltex/docs/commands.html#ltex-activate-extension). However, without a parser, this will yield some false positives regarding org-mode syntax.

**Comment by sirno** [comment]
Thanks @valentjn for the effort!

**Comment by valentjn** [comment]
If the result doesn't have to be 100% accurate (regarding false positives), I'll try writing a parser myself. It doesn't have to be an actual parser; it suffices to strip all the markup away. If the syntax of org-mode is reasonable (which I don't know), this might be within my reach.

**Comment by sirno** [comment]
Since it is very similar to markdown, it may even be possible to adjust a md parser to make it work. For me, being able to activate the extension to quickly check text in all sorts of places is already great.

**Comment by valentjn** [comment]
I finished the Org parser. Now I know enough Org to be able to say that Org isn't sufficiently similar to Markdown. Reusing the Markdown parser was not an option, unfortunately.

This is fixed in 10.1.0.
